app/services/PushPullNotificationService.scala

```diff
@@ -47,17 +54,30 @@ class PushPullNotificationService @Inject()(connector: PushPullNotificationConne
           None
       }
 
-  def sendPushNotification(boxId: BoxId, notification: ArrivalMessageNotification)(implicit ec: ExecutionContext, hc: HeaderCarrier): Future[Unit] =
-    connector
-      .postNotification(boxId, notification)
+  private[services] def sendPushNotification(inboundRequest: InboundMessageRequest, headers: Headers)(implicit hc: HeaderCarrier): Future[Unit] =
+    inboundRequest.arrival.notificationBox
       .map {
-        case Left(UpstreamErrorResponse(message, statusCode, _, _)) =>
-          logger.warn(s"Error $statusCode received while sending notification for boxId $boxId: $message")
-        case Right(_) => ()
+        box =>
+          val contentLength = headers
+            .get(HeaderNames.CONTENT_LENGTH)
+            .flatMap(
+              x => Try(x.toInt).toOption
+            )
 
+          val arrivalMessageNotification = ArrivalMessageNotification.fromInboundRequest(inboundRequest, contentLength)
+
+          connector
+            .postNotification(box.boxId, arrivalMessageNotification)
+            .map {
+              case Left(UpstreamErrorResponse(message, statusCode, _, _)) =>
+                logger.warn(s"Error $statusCode received while sending notification for boxId ${box.boxId}: $message")
+              case Right(_) => ()
+
+            }
+            .recover {
+              case NonFatal(e) =>
+                logger.error(s"Error while sending push notification", e)
+            }
```

Comment:
It does feel a little bit like we are moving some HTTP-level logic (what I'd usually consider controller stuff) into the service layer here - not a comment that needs to be addressed but maybe something we need to think about

Reply:
All of the logic is the same from before, except instead of passing in the values boxId and notification, we're making this service dependent on needing a InboundMessageRequest which has those bits of data in it or has the data which we should work out in this service (eg using `ArrivalMessageNotification.fromInboundRequest`). 

Is that the main concern that we're passing in the headers to this service layer?

Reply:
Yeah I think so, but let's leave as is for now as like I say it's not a huge problem